flight/PiOS/STM32/chibi_main.c

```diff
@@ -139,6 +141,19 @@ void initTask(void)
 	AlarmsInitialize();
 #endif
 
+	uint8_t serial[PIOS_SYS_SERIAL_NUM_BINARY_LEN];
+	PIOS_SYS_SerialNumberGetBinary(serial);
+
+	for (int i = 0; i < PIOS_SYS_SERIAL_NUM_BINARY_LEN; i += 4) {
+		randomize_addseed(
+				(serial[i]) |
+				(serial[i+1] << 8) |
+				(serial[i+2] << 16) |
+				(serial[i+2] << 24));
+	}
+
+	/* TODO: consider mixing in HRNG on F4 */
```

Comment:
Accel/gyro data would be a good source of entropy too, just not available this early. 😀

Reply:
Yes-- the problem is the RNG is not reentrant.  So once there's tasks it's not really safe to seed it anymore.

Reply:
Ack. You would have to do it in board_init after gyro driver is init'ed.

Reply:
Turns out SRAM can be a good source of entropy if we want a secure PRNG: https://eprint.iacr.org/2013/304.pdf

Reply:
Could be-- though lots of things init it etc by this point.